features/cellular/framework/AT/AT_CellularDevice.cpp

```diff
@@ -176,7 +176,14 @@ nsapi_error_t AT_CellularDevice::set_pin(const char *sim_pin)
 
     _at->lock();
     _at->cmd_start("AT+CPIN=");
+
+    const bool stored_debug_state = _at->get_debug();
+    _at->set_debug(false);
+
     _at->write_string(sim_pin);
```

Comment:
Is there a reason that this line wasn't instead removed?

Reply:
Not sure what you are referring here as injected code:
1) stores current tracing state (enable/disable)
2) disable tracing
3) resumes stored tracing state after sim pin has been written to the modem (cellular subsystem)